NYTPhotoViewer/NYTPhotosViewController.m

```diff
@@ -275,7 +283,7 @@ - (void)actionButtonTapped:(id)sender {
     }
     
     if (!clientDidHandle && (self.currentlyDisplayedPhoto.image || self.currentlyDisplayedPhoto.imageData)) {
-        UIImage *image = self.currentlyDisplayedPhoto.image ? self.currentlyDisplayedPhoto.image : [UIImage imageWithData:self.currentlyDisplayedPhoto.imageData];
+        NSData *image = self.currentlyDisplayedPhoto.image ? UIImagePNGRepresentation(self.currentlyDisplayedPhoto.image) : self.currentlyDisplayedPhoto.imageData;
```

Comment:
Could this be `id image =` and allow either the `UIImage` object or the `NSData` object to be passed to the activity VC? That way you don't have to use `UIImagePNGRepresentation()`

Reply:
Yes @joewalsh ! Thanks for the information, I will add it right now.